neo4j/src/main/java/org/hibernate/ogm/datastore/neo4j/remote/common/request/impl/RemoteStatements.java

```diff
@@ -0,0 +1,35 @@
+/*
+ * Hibernate OGM, Domain model persistence for NoSQL datastores
+ *
+ * License: GNU Lesser General Public License (LGPL), version 2.1 or later
+ * See the lgpl.txt file in the root directory or <http://www.gnu.org/licenses/lgpl-2.1.html>.
+ */
+package org.hibernate.ogm.datastore.neo4j.remote.common.request.impl;
+
+import java.util.ArrayList;
+import java.util.Collections;
+import java.util.Iterator;
+import java.util.List;
+
+/**
+ * Collects a group of {@link RemoteStatement}.
+ *
+ * @author Davide D'Alto
+ */
+public class RemoteStatements implements Iterable<RemoteStatement> {
+
+	private final List<RemoteStatement> statements = new ArrayList<>();
+
+	public void add(RemoteStatement statement) {
+		statements.add( statement );
+	}
+
+	@Override
+	public Iterator<RemoteStatement> iterator() {
+		return Collections.unmodifiableCollection( statements ).iterator();
+	}
+
+	public void addStatement(RemoteStatement statement) {
+		statements.add( statement );
```

Comment:
Why is there a `add` and a `addStatement`? Looks like we should remove one. Btw, maybe we could return `this` so that we could chain the calls?